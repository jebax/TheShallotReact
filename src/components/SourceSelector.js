import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class SourceSelector extends Component {
  state = {
    selected: 'BBC News'
  }

  handleChange = event => {
    const index = event.target.selectedIndex
    const selected = event.target[index].text

    this.setState({ selected: selected })
    this.props.changeFunction(event)
  }

  render() {
    return (
      <select
        selected={this.state.selected}
        onChange={this.handleChange}
        className='sourceSelector'
      >
        <option
          value='bbc-news'
        >
          BBC News
        </option>
        <option
          value='the-guardian-uk'
        >
          The Guardian
        </option>
        <option
          value='financial-times'
        >
          Financial Times
        </option>
        <option
          value='google-news'
        >
          Google News
        </option>
        <option
          value='national-geographic'
        >
          National Geographic
        </option>
        <option
          value='techradar'
        >
          TechRadar
        </option>
        <option
          value='time'
        >
          Time
        </option>
        <option
          value='vice-news'
        >
          Vice News
        </option>
        <option
          value='wired'
        >
          Wired
        </option>
      </select>
    )
  }
}

SourceSelector.propTypes = {
  changeFunction: PropTypes.func.isRequired
}
