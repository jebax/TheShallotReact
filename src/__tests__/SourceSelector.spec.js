import React from 'react'
import { shallow, mount } from 'enzyme'
import SourceSelector from '../components/SourceSelector'

describe('<SourceSelector />', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(
      <SourceSelector
        changeFunction={() => {}}
      />
    )
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a select element', () => {
    expect(wrapper.find('select').length).toEqual(1)
  })

  it('should have multiple select options', () => {
    const selector = wrapper.find('select')

    selector.simulate('click')
    expect(wrapper.find('option').length).toBeGreaterThan(1)
  })

  it('should default to BBC News, and have BBC as first option', () => {
    const selector = wrapper.find('select')

    expect(selector.props().selected).toEqual('BBC News')

    selector.simulate('click')

    const firstOption = wrapper.find('option').first()
    expect(firstOption.text()).toEqual('BBC News')
  })
})
