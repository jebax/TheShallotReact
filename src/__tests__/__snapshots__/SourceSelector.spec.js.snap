// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SourceSelector /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SourceSelector
    changeFunction={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          value="bbc-news"
        >
          BBC News
        </option>,
        <option
          value="the-guardian-uk"
        >
          The Guardian
        </option>,
        <option
          value="financial-times"
        >
          Financial Times
        </option>,
        <option
          value="google-news"
        >
          Google News
        </option>,
        <option
          value="national-geographic"
        >
          National Geographic
        </option>,
        <option
          value="techradar"
        >
          TechRadar
        </option>,
        <option
          value="time"
        >
          Time
        </option>,
        <option
          value="vice-news"
        >
          Vice News
        </option>,
        <option
          value="wired"
        >
          Wired
        </option>,
      ],
      "className": "sourceSelector",
      "onChange": [Function],
      "selected": "BBC News",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BBC News",
          "value": "bbc-news",
        },
        "ref": null,
        "rendered": "BBC News",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Guardian",
          "value": "the-guardian-uk",
        },
        "ref": null,
        "rendered": "The Guardian",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Financial Times",
          "value": "financial-times",
        },
        "ref": null,
        "rendered": "Financial Times",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Google News",
          "value": "google-news",
        },
        "ref": null,
        "rendered": "Google News",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "National Geographic",
          "value": "national-geographic",
        },
        "ref": null,
        "rendered": "National Geographic",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TechRadar",
          "value": "techradar",
        },
        "ref": null,
        "rendered": "TechRadar",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Time",
          "value": "time",
        },
        "ref": null,
        "rendered": "Time",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vice News",
          "value": "vice-news",
        },
        "ref": null,
        "rendered": "Vice News",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Wired",
          "value": "wired",
        },
        "ref": null,
        "rendered": "Wired",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="bbc-news"
          >
            BBC News
          </option>,
          <option
            value="the-guardian-uk"
          >
            The Guardian
          </option>,
          <option
            value="financial-times"
          >
            Financial Times
          </option>,
          <option
            value="google-news"
          >
            Google News
          </option>,
          <option
            value="national-geographic"
          >
            National Geographic
          </option>,
          <option
            value="techradar"
          >
            TechRadar
          </option>,
          <option
            value="time"
          >
            Time
          </option>,
          <option
            value="vice-news"
          >
            Vice News
          </option>,
          <option
            value="wired"
          >
            Wired
          </option>,
        ],
        "className": "sourceSelector",
        "onChange": [Function],
        "selected": "BBC News",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BBC News",
            "value": "bbc-news",
          },
          "ref": null,
          "rendered": "BBC News",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Guardian",
            "value": "the-guardian-uk",
          },
          "ref": null,
          "rendered": "The Guardian",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Financial Times",
            "value": "financial-times",
          },
          "ref": null,
          "rendered": "Financial Times",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Google News",
            "value": "google-news",
          },
          "ref": null,
          "rendered": "Google News",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "National Geographic",
            "value": "national-geographic",
          },
          "ref": null,
          "rendered": "National Geographic",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TechRadar",
            "value": "techradar",
          },
          "ref": null,
          "rendered": "TechRadar",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Time",
            "value": "time",
          },
          "ref": null,
          "rendered": "Time",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vice News",
            "value": "vice-news",
          },
          "ref": null,
          "rendered": "Vice News",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Wired",
            "value": "wired",
          },
          "ref": null,
          "rendered": "Wired",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
